lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -363,15 +420,14 @@ def sample_from_image(self, num_pixels, reduce=np.sum, *, correct_origin=None):
 
         # Time and coordinates are being cast to an integer since we use them to index into a data
         # array. Note that coordinate pixel centers are defined at integer coordinates.
-        offset = 0.0 if not correct_origin else 0.5
-        return [
-            reduce(
-                self._image[
-                    max(int(c + offset) - num_pixels, 0) : int(c + offset) + num_pixels + 1, int(t)
-                ]
-            )
-            for t, c in zip(self.time_idx, self.coordinate_idx)
-        ]
+        return _sample_from_image(
+            self._image,
+            num_pixels,
+            self.time_idx,
+            self.coordinate_idx,
+            reduce=reduce,
+            correct_origin=correct_origin,
+        )
```

Comment:
I was thinking about that as I was preparing this PR. One property that is sort of nice now, is that you always know that the photon counts you get originate from the same method (with the same algorithm parameters) as was used for the refinement procedure. If you'd add a `resample_photon_counts()`, you would lose that. You'd end up with a count that does not necessarily correspond to the same parameters used in localization.

It's kind of weird to pack subpixel position data for a window size of `N=5` with a photon count acquired with a window size of `N=10` for example. I feel that may lead to subtle issues in the future.

If we decide that, then maybe it would be better to _not_ pack the photon counts in with the localization itself, but have a dedicated dataclass for counts. What do you think?

Reply:
yeah I had also been thinking it'd be weird to not have a way of knowing what window was used for sampling. The other thing to worry about is the exported CSV, I think we just use a default value now so we need to make sure we don't break that format. Maybe it's best to not deprecate in this PR and have a more general discussion about it during refinement